packages/core/src/utils/parseInvitation.ts

```diff
@@ -105,6 +108,33 @@ export const oobInvitationFromShortUrl = async (response: Response): Promise<Out
   throw new AriesFrameworkError('HTTP request time out or did not receive valid response')
 }
 
+export function transformLegacyConnectionlessInvitationToOutOfBandInvitation(messageJson: Record<string, unknown>) {
+  const agentMessage = JsonTransformer.fromJSON(messageJson, AgentMessage)
+
+  // ~service is required for legacy connectionless invitations
+  if (!agentMessage.service) {
+    throw new AriesFrameworkError('Invalid legacy connectionless invitation url. Missing ~service decorator.')
+  }
+
+  // This destructuring removes the ~service property from the message, and
+  // we can can use messageWithoutService to create the out of band invitation
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { '~service': service, ...messageWithoutService } = messageJson
+
+  // transform into out of band invitation
+  const invitation = new OutOfBandInvitation({
+    // The label is currently required by the OutOfBandInvitation class, but not according to the specification.
+    // FIXME: In 0.5.0 we will make this optional: https://github.com/hyperledger/aries-framework-javascript/issues/1524
+    label: '',
```

Comment:
If we can merge the optional label PR first, we can simply avoid using this empty label. See https://github.com/openwallet-foundation/credo-ts/pull/1680#issuecomment-1914514095

Reply:
I'll update once #1680 is merged